### PR TITLE
Add Test for __OW_NAMESPACE Environment Variable

### LINKED ISSUE #1664: Environment variables only provides namespace of owner of action
Steps to reproduce:

If you create a package and shared it let's say `/whisk.system/carlos_experiments`
And create an action `env` with fqname `/whisk.system/carlos_experiments/env`

Create shared package
```
wsk package update carlos_experiments --shared yes
```

Giving a function `env.js`
```
$ cat env.js
function main(){
return process.env;
}
```
Create action env
```
$ wsk action update /whisk.system/carlos_experiments/env env.js 
ok: updated action carlos_experiments/env
```

List packages
```
$ wsk list
Entities in namespace: default
packages
/whisk.system/carlos_experiments                                       shared
```
List summary of package
```
$ wsk package get carlos_experiments --summary
package /whisk.system/carlos_experiments
 action /whisk.system/carlos_experiments/env
```

If now I invoke from a different namespace other than `whisk.system` like `carlos@cool.com_dev`
```
$ wsk action invoke /whisk.system/carlos_experiments/env -b -r
{
    "DEBIAN_FRONTEND": "noninteractive",
    "HOME": "/root",
    "HOSTNAME": "d29e52600f60",
    "NODE_VERSION": "6.9.1",
    "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
    "PWD": "/nodejsAction",
    "SERVICE_IGNORE": "true",
    "__OW_ACTION_NAME": "/whisk.system/carlos_experiments/env",
    "__OW_ACTIVATION_ID": "50e739cb3ab24",
    "__OW_API_HOST": "10.155.229.130:443",
    "__OW_API_KEY": "fa59:zILraFhLbDd5yWQ",
    "__OW_DEADLINE": "1481932398872",
    "__OW_NAMESPACE": "whisk.system"
}
```
As a feed provider implementator I need a namespace for the user invoking the action, not the namespace of the feed action.
This is required to store the trigger data in a DB to create a unique id in the DB, so user can in a later time when it deletes the trigger the feed action will be able to identify the trigger to delete

Here is the use case creating a trigger using a feed action
```
$ wsk trigger create myt -f /whisk.system/carlos_experiments/env
ok: invoked /whisk.system/carlos_experiments/env with id 7f4676509
{
    "namespace": "carlos@cool.com_dev",
    "name": "env",
    "version": "0.0.1",
    "subject": "carlos@cool.com",
    "activationId": "7f4676509",
    "start": 1481931584737,
    "end": 1481931584795,
    "duration": 58,
    "response": {
        "status": "success",
        "statusCode": 0,
        "success": true,
        "result": {
            "DEBIAN_FRONTEND": "noninteractive",
            "HOME": "/root",
            "HOSTNAME": "38001e3b19cb",
            "NODE_VERSION": "6.9.1",
            "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "PWD": "/nodejsAction",
            "SERVICE_IGNORE": "true",
            "__OW_ACTION_NAME": "/whisk.system/carlos_experiments/env",
            "__OW_ACTIVATION_ID": "7f4676509",
            "__OW_API_HOST": "10.155.229.130:443",
            "__OW_API_KEY": "fa59:zILraFhLbDd5yWQ",
            "__OW_DEADLINE": "1481931644784",
            "__OW_NAMESPACE": "whisk.system"
        }
    },
    "logs": [],
    "annotations": [
        {
            "key": "limits",
            "value": {
                "logs": 10,
                "memory": 256,
                "timeout": 60000
            }
        },
        {
            "key": "path",
            "value": "whisk.system/carlos_experiments/env"
        }
    ],
    "publish": false
}
ok: created trigger myt
``` 

As you can see the value for `__OW_NAMESPACE` is always `whisk.system`

Can we add a new environment variable to get the value of the namespace associated to the authkey being use to invoke the action or feed action, in this case `carlos@cool.com_dev`

Maybe `"__OW_NAMESPACE_INVOKE":"carlos@cool.com_dev"`

cc @bjustin-ibm @jasonpet @dubeejw @johnsolm

### PR DESCRIPTION
**Pull request by dubeejw**
- Ensure that __OW_NAMESPACE returns the subject's namespace when invoking an action not owned by the subject

Fixes #1664.

### CODE REVIEW

File: tests/dat/actions/namespace.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function main() {
```
Comment (rabbah) [inline tests/dat/actions/namespace.js]: can you reuse the helloContext.js action file?

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 2]
```diff
@@ -323,6 +324,27 @@ class WskEntitlementTests
...
+                (pkg, _) => pkg.create(samplePackage, shared = Some(true))(wp)
+            }
+
+            assetHelper.withCleaner(wsk.action, fullSampleActionName) {
+                val file = Some(TestUtils.getTestActionFilename("namespace.js"))
+                (action, _) => action.create(fullSampleActionName, file)(wp)
+            }
+
+            val fullyQualifiedActionName = s"/$guestNamespace/$fullSampleActionName"
+            val run = wsk.action.invoke(fullSampleActionName)(guestWskProps)
```
Comment (rabbah) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: I can't tell from this test alone (have to look at the values of `samplePackage` and `fullSampleActionName`
if the invoke is the desired one.

The test needs to invoke with key `A` an action in namespace `B`. The invoke here is using `guestWskProps` so let's call that key `A`. The action hence needs to be in namespace `B` - I don't think `fullSampleActionName` or `fullyQualifiedActionName` are correct in that case.

**Comment by dubeejw** [comment]
@rabbah, I made updates to make it clearer as to what is happening in the test. I am creating a packaged action with the namespace and auth key that is provided by the default WskProps. From there I am creating a guest user and invoking the previously created action with the guest user's namespace and auth key.

**Comment by dubeejw** [comment]
@rabbah, Travis keeps failing due to the job talking to long. PG2 1665 is approved.

**Comment by rabbah** [comment]
Cherry picked the test into my branch and made a couple of small tweaks.

**Pull request closed without merging**